services/addons/images/firmware_manager/commsignia_upgrader.py

```diff
@@ -0,0 +1,70 @@
+from paramiko import SSHClient, WarningPolicy
+from scp import SCPClient
+import upgrader
+import json
+import logging
+import os
+import sys
+
+class CommsigniaUpgrader( upgrader.UpgraderAbstractClass ):
+  def __init__(self, upgrade_info):
+    super().__init__(upgrade_info)
+
+  def upgrade(self):
+    try:
+      # Download firmware installation package
+      self.download_blob()
+
+      # Make connection with the target device
+      logging.info("Making SSH connection with the device...")
+      ssh = SSHClient()
+      ssh.set_missing_host_key_policy(WarningPolicy)
+      ssh.connect(self.rsu_ip, username=self.ssh_username, password=self.ssh_password, look_for_keys=False, allow_agent=False)
+
+      # Make SCP client to copy over the firmware installation package to the /tmp/ directory on the remote device
+      logging.info("Copying installation package to the device...")
+      scp = SCPClient(ssh.get_transport())
+      scp.put(self.local_file_name, remote_path='/tmp/')
+      scp.close()
+
+      # Delete local installation package and its parent directory so it doesn't take up storage space
+      self.cleanup()
+
+      # Run firmware upgrade and reboot
+      logging.info("Running firmware upgrade...")
+      _stdin, _stdout,_stderr = ssh.exec_command(f"signedUpgrade.sh /tmp/{self.install_package}")

```

Comment:
Where is this `signedUpgrade.sh` file from?

Reply:
This is actually a native script on all Commsignia devices

Reply:
Ok, that's what I was assuming but thought I'd check since there wasn't a reference to it anywhere else here.